app/src/main/java/vn/sun/music_68/data/repository/TrackRepository.java

```diff
@@ -0,0 +1,63 @@
+package vn.sun.music_68.data.repository;
+
+import vn.sun.music_68.data.model.Track;
+import vn.sun.music_68.data.source.TrackDataSource;
+
+public class TrackRepository implements TrackDataSource.Local, TrackDataSource.Remote {
+    private static TrackRepository sInstance;
+    private TrackDataSource.Local mLocalDataSource;
+    private TrackDataSource.Remote mRemoteDataSource;
+
+    public static synchronized TrackRepository getInstance(TrackRepository.Remote remoteDataSource,
+                                                           TrackRepository.Local localDataSource) {
+        if (sInstance == null) {
```

Comment:
```suggestion
      public static SingletonClass getInstance() {
        if (SINGLE_INSTANCE == null) {
            synchronized (SingletonClass.class) {
                if (SINGLE_INSTANCE == null) {
                    SINGLE_INSTANCE = new SingletonClass();
                }
            }
        }
        return SINGLE_INSTANCE;
    }
```
tham khảo nhé:https://dzone.com/articles/singleton-in-java
cái này không tối ưu vì mỗi lần getInstance nó sẽ synchronized, gây tốn tài nguyên